### ISSUE TITLE
On Apple Silicon, `Error relocating /lib64/libz.so.1: ...symbol not found`

### ISSUE THREAD

**Issue by giuscri** [yt-dlp/yt-dlp#10072]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

When using `2024.05.27` release, if I run `yt-dlp` from an amd64 Docker container of `public.ecr.aws/lambda/python:3.12` the binary won't start properly because of libz issues. Please note that previous releases of `yt-dlp` (e.g. the April one) worked fine on the same setup.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
$ docker run --platform=linux/amd64 --entrypoint=bash -ti --rm public.ecr.aws/lambda/python:3.12 
bash-5.2# curl -sLO https://github.com/yt-dlp/yt-dlp/releases/download/2024.05.27/yt-dlp_linux
bash-5.2# chmod +x ./yt-dlp_linux 
bash-5.2# ./yt-dlp_linux --help
Error relocating /lib64/libz.so.1: crc32_z: symbol not found
Error relocating /lib64/libz.so.1: __snprintf_chk: symbol not found
Error relocating /lib64/libz.so.1: __vsnprintf_chk: symbol not found
bash-5.2# ./yt-dlp_linux -vU --help
Error relocating /lib64/libz.so.1: crc32_z: symbol not found
Error relocating /lib64/libz.so.1: __snprintf_chk: symbol not found
Error relocating /lib64/libz.so.1: __vsnprintf_chk: symbol not found
bash-5.2# curl -sLO https://github.com/yt-dlp/yt-dlp/releases/download/2024.04.09/yt-dlp_linux
bash-5.2# ./yt-dlp_linux -vU --help | head
[debug] Command-line config: ['-vU', '--help']

Usage: yt-dlp_linux [OPTIONS] URL [URL...]

Options:
  General Options:
    -h, --help                      Print this help text and exit
    --version                       Print program version and exit
    -U, --update                    Update this program to the latest stable
                                    version
    --no-update                     Do not check for updates (default)
    --update-to [CHANNEL]@[TAG]     Upgrade/downgrade to a specific version.
```

**Comment by bashonly** [comment]
Are ffmpeg/ffprobe/phantomjs installed in the container? Does `yt-dlp_linux --version` work? (without `-v`)

**Comment by bashonly** [comment]
Was able to reproduce the issue with the same docker image. However, it worked fine with a basic ubuntu image and a basic alpine image (both amd64).

**Comment by rihadavid** [comment]
This is also happening to me on AWS lambda layer, x86_64, Node.js18.
It starts with [yt-dlp 2024.05.26](https://github.com/yt-dlp/yt-dlp/releases/tag/2024.05.26).
Earlier versions are broken with 403 error on youtube, unfortunately, so I can't use any build.

**Comment by rihadavid** [comment]
I was able to pinpoint the origin furhter to release [yt-dlp master 2024.04.28.221944](https://github.com/yt-dlp/yt-dlp-master-builds/releases/tag/2024.04.28.221944) and [yt-dlp nightly 2024.04.28.232723](https://github.com/yt-dlp/yt-dlp-nightly-builds/releases/tag/2024.04.28.232723). Before this, it was not happening. 

I suspect it might be because of the "[Migrate linux_exe to static musl builds](https://github.com/yt-dlp/yt-dlp/commit/ac817bc83efd939dca3e40c4b527d0ccfc77172b)"

**Comment by bashonly** [comment]
Yeah, our conda-based linux build flow was broken due to reasons unknown, and we had been wanting to move away from conda for a long time anyways. The easiest way to keep compat with most x86_64 linux systems was static musllibc builds

**Comment by rihadavid** [comment]
Is there anything I can do to make this work in aws lambda?

**Comment by rihadavid** [comment]
I managed to solve it by including `libz.so.1` from [previous zip build](https://github.com/yt-dlp/yt-dlp-master-builds/releases/download/2024.04.21.224157/yt-dlp_linux.zip) in my lambda layer, in subfolder `/lib` and setting environment variable `LD_LIBRARY_PATH` to `/opt/lib`

**Comment by History-exe** [comment]
ubuntu 24.04
yt-dlp 2025.07.21
Use yt_dlp download Youtube video, show as below:
```
Error relocating /usr/lib/x86_64-linux-gnu/libz.so.1: __snprintf_chk: symbol not found
Error relocating /usr/lib/x86_64-linux-gnu/libz.so.1: __vsnprintf_chk: symbol not found
```
CPU: AMD Ryzen 5 5500U

**Comment by KingMob** [comment]
I'm seeing the exact same error as History-exe in my Docker build.

Image: eclipse-temurin:17-jdk-noble, which is based off Ubuntu 24.04
[yt-dlp 2025.07.21](https://github.com/yt-dlp/yt-dlp/releases/tag/2025.07.21) (downloaded as yt-dlp_linux)

```
#29 [stage-2 14/14] RUN yt-dlp --version
#29 1.229 Error relocating /usr/local/lib/libz.so.1: __snprintf_chk: symbol not found
#29 1.229 Error relocating /usr/local/lib/libz.so.1: __vsnprintf_chk: symbol not found
#29 ERROR: process "/bin/sh -c yt-dlp --version" did not complete successfully: exit code: 127
```

----

As a side note, I'm running an Intel Mac, so this issue is broader than the description suggests.

**Comment by History-exe** [comment]
Thanks for reply and tag the solution. Now I use the python version of yt_dlp, it works.

For someone don't know how, just use this as usual:

1. download the [yt-dlp python ver](https://github.com/yt-dlp/yt-dlp/releases/download/2025.08.11/yt-dlp)
2. type this in terminal.
    `python3 -m yt_dlp`

**Comment by andrade** [comment]
I'm having the same issue when using `torsocks`.

- Works: `$ yt-dlp https://www.youtube.com/watch?v=BneUk3669BA`
- Fails: `$ torsocks yt-dlp https://www.youtube.com/watch?v=BneUk3669BA` (it worked with a previous version of yt-dlp).

```
$ torsocks yt-dlp https://www.youtube.com/watch?v=BneUk3669BA
Error relocating /usr/lib/x86_64-linux-gnu/torsocks/libtorsocks.so: __isoc23_strtoul: symbol not found
Error relocating /usr/lib/x86_64-linux-gnu/torsocks/libtorsocks.so: __snprintf_chk: symbol not found
Error relocating /usr/lib/x86_64-linux-gnu/torsocks/libtorsocks.so: __fdelt_chk: symbol not found
Error relocating /usr/lib/x86_64-linux-gnu/torsocks/libtorsocks.so: __memcpy_chk: symbol not found
Error relocating /usr/lib/x86_64-linux-gnu/torsocks/libtorsocks.so: __isoc23_strtol: symbol not found
Error relocating /usr/lib/x86_64-linux-gnu/torsocks/libtorsocks.so: __vsnprintf_chk: symbol not found
Error relocating /usr/lib/x86_64-linux-gnu/torsocks/libtorsocks.so: __fprintf_chk: symbol not found
```

My software versions:
```
$ torsocks --version
Torsocks 2.3.0

$ yt-dlp --version
2025.08.11

$ lsb_release -d
Description:	Ubuntu 25.04
```

---

Similarly to @rihadavid, I started having problems after the migration to static musl builds.

- [yt-dlp 2024.04.09](https://github.com/yt-dlp/yt-dlp/releases/tag/2024.04.09) still works, but
- [yt-dlp 2024.05.26](https://github.com/yt-dlp/yt-dlp/releases/tag/2024.05.26) fails with the error above.
